drivers/timer/cortex_m_systick.c

```diff
@@ -260,85 +309,95 @@ void sys_clock_set_timeout(int32_t ticks, bool idle)
 	}
 #endif /* CONFIG_CORTEX_M_SYSTICK_IDLE_TIMER */
 
-#if defined(CONFIG_TICKLESS_KERNEL)
-	uint32_t delay;
-	uint32_t val1, val2;
-	uint32_t last_load_ = last_load;
+	/* passing ticks==1 means "announce the next tick",
+	 * ticks value of zero (or even negative) is legal and
+	 * treated identically: it simply indicates the
+	 * kernel would like the next tick announcement ASAP.
+	 */
+	ticks = CLAMP(ticks - 1, 0, (const int32_t)max_ticks);
 
-	ticks = (ticks == K_TICKS_FOREVER) ? MAX_TICKS : ticks;
-	ticks = CLAMP(ticks - 1, 0, (int32_t)MAX_TICKS);
+	K_SPINLOCK(&lock) {
+		uint32_t delay;
+		uint32_t val1, val2;
+		uint32_t last_load_ = last_load;
 
-	k_spinlock_key_t key = k_spin_lock(&lock);
+		uint32_t pending = elapsed();
 
-	uint32_t pending = elapsed();
+		val1 = sys_read32(SYST_CVR);
 
-	val1 = SysTick->VAL;
+		cycle_count += pending;
+		overflow_cyc = 0;
 
-	cycle_count += pending;
-	overflow_cyc = 0U;
+		uint32_t unannounced = cycle_count - announced_cycles;
 
-	uint32_t unannounced = cycle_count - announced_cycles;
-
-	if ((int32_t)unannounced < 0) {
-		/* We haven't announced for more than half the 32-bit
-		 * wrap duration, because new timeouts keep being set
-		 * before the existing one fires.  Force an announce
-		 * to avoid loss of a wrap event, making sure the
-		 * delay is at least the minimum delay possible.
-		 */
-		last_load = MIN_DELAY;
-	} else {
-		/* Desired delay in the future */
-		delay = ticks * CYC_PER_TICK;
-
-		/* Round delay up to next tick boundary */
-		delay += unannounced;
-		delay = DIV_ROUND_UP(delay, CYC_PER_TICK) * CYC_PER_TICK;
-		delay -= unannounced;
-		delay = MAX(delay, MIN_DELAY);
-		if (delay > MAX_CYCLES) {
-			last_load = MAX_CYCLES;
+		if ((int32_t)unannounced < 0) {
+			/* We haven't announced for more than half the 32-bit
+			 * wrap duration, because new timeouts keep being set
+			 * before the existing one fires.  Force an announce
+			 * to avoid loss of a wrap event, making sure the
+			 * delay is at least the minimum delay possible.
+			 */
+			last_load = MIN_DELAY;
 		} else {
-			last_load = delay;
+			const uint32_t max_cycles = max_ticks * CYC_PER_TICK;
```

Comment:
IMHO it makes it clearer that this is a constant to have this as an all CAPS `#define` rather than a variable looking snake-case `const`.

Reply:
I've not objections. Accepting.